public/app/core/angular_wrappers.ts

```diff
@@ -196,4 +197,11 @@ export function registerAngularDirectives() {
     ['onNowDelayChange', { watchDepth: 'reference', wrapApply: true }],
     ['onHideTimePickerChange', { watchDepth: 'reference', wrapApply: true }],
   ]);
+
+  react2AngularDirective('azureMonitorQueryEditor', QueryEditor, [
+    'query',
+    ['datasource', { watchDepth: 'reference' }],
+    'replaceTemplateVariable',
+    'onQueryChange',
+  ]);
```

Comment:
...and add onRunQuery. You should be able to run onChange without triggering a panel refresh